tests/unit/s2n_fingerprint_ja4_test.c

```diff
@@ -731,32 +731,62 @@ int main(int argc, char **argv)
 
                 EXPECT_TRUE(output_size > S2N_JA4_A_ALPN_LAST);
                 EXPECT_EQUAL(output[S2N_JA4_A_ALPN_FIRST], 'q');
-                EXPECT_EQUAL(output[S2N_JA4_A_ALPN_LAST], '0');
+                EXPECT_EQUAL(output[S2N_JA4_A_ALPN_LAST], 'q');
             };
 
-            /* Test non-ascii alpn value */
+            /* Test non-ascii alpn values
+             *
+             * The spec does not currently define this case, but will be updated in the
+             * future according to https://github.com/FoxIO-LLC/ja4/issues/148
+             */
             {
+                struct {
+                    const uint8_t bytes[4];
+                    const size_t bytes_size;
+                    const char *str;
+                } test_cases[] = {
+                    { .bytes = { 0xAB }, .str = "ab", .bytes_size = 1 },
+                    { .bytes = { 0xAB, 0xCD }, .str = "ad", .bytes_size = 2 },
+                    { .bytes = { 0x30, 0xAB }, .str = "3b", .bytes_size = 2 },
+                    { .bytes = { 0x30, 0x31, 0xAB, 0xCD }, .str = "3d", .bytes_size = 4 },
+                    { .bytes = { 0x30, 0xAB, 0xCD, 0x31 }, .str = "01", .bytes_size = 4 },
```

Comment:
I might be missing something - but why shouldn't `str` be `31` in this test?

Reply:
Ohhh these are valid ascii...

Reply:
Yeah I think that'll be clearer when the spec is finalized and I can add a duvet comment.